src/OSLikeStuff/task_scheduler.cpp

```diff
@@ -100,34 +103,48 @@ TaskID TaskManager::chooseBestTask(double deadline) {
 			}
 		}
 	}
+
 	return bestTask;
 }
 
 TaskID TaskManager::addRepeatingTask(TaskHandle task, uint8_t priority, double minTimeBetweenCalls,
                                      double targetTimeBetweenCalls, double maxTimeBetweenCalls) {
-	list[index] = (Task{task, priority, 0, 0, minTimeBetweenCalls, targetTimeBetweenCalls, maxTimeBetweenCalls});
+	list[index] = (Task{task, priority, 0, 0, minTimeBetweenCalls, targetTimeBetweenCalls, maxTimeBetweenCalls, false});
+	createSortedList();
+	return index++;
+}
+
+TaskID TaskManager::addOnceTask(TaskHandle task, uint8_t priority, double timeToWait) {
```

Comment:
do we want to add some sanity checks of re-adding an already added once task?

Reply:
I think it might be useful to schedule the same task to run once at a couple different times